### ISSUE TITLE
Arithmetic error in BasicGUI.cpp

### ISSUE THREAD

**Issue by desttinghim** [enigma-dev/enigma-dev#1031]
```
int gui_style_create(){
        gui::gui_data_elements.emplace(gui::gui_data_elements_maxid, gui::Style());
    gui::Style &sty = gui::gui_data_elements[gui::gui_data_elements_maxid];
        sty.id = gui::gui_data_elements_maxid;
    //printf("Creating style with size %i\n", sizeof(gui::gui_data_elements[gui::gui_data_elements_maxid]));
        return gui::gui_data_elements_maxid++;
    }
```

This piece of code seems to be failing. Specifically, the emplace command is triggering an Arithmetic error. It looks like a modulus is being done, and that is causing a divide by zero error.

```
 “/tmp/egm1633959076788919710.tmp” terminated by signal SIGFPE (Floating point exception)
```

This is the error I get when I manually run the compiled code from the command line. This problem appears to happen with every project. I'm not sure what I'm doing wrong.

**Comment by faissaloo** [comment]
This is an issue is a hashing bug in the BasicGUI extension: https://github.com/enigma-dev/enigma-dev/pull/929#issuecomment-240378999  
For now you can disable it by going to game settings->extensions and untic basicGUI

**Comment by desttinghim** [comment]
Alright, that worked. Took a little bit to find that. What's the difference between the green puzzle piece extensions and the ones in game settings?

**Comment by faissaloo** [comment]
I believe the green puzzle piece is meant to be for GEX extensions but I don't think they were ever implemented, @RobertBColton please correct me if I'm wrong.

**Comment by desttinghim** [comment]
Hmm. Maybe the BasicGUI extension should be disabled by default until the problem is resolved, because it's difficult to tell what is going on from within LateralGM. To me, it just looked like my code was not being run until I dug a little deeper.

**Comment by RobertBColton** [comment]
Harri changed it way back to be disabled by default as you suggested in f7c5b11ecbbcd20dd171896548cd44625bcfa96e

Even when I do enable the extension now running an empty game from emake I do not get the arithmetic error. It doesn't occur running the output exe in gdb either.
```bash
./emake -o /tmp/test.exe -r -e "BasicGUI"
```

I had a hard time deciphering what you guys were talking about here until I scrolled up a bit from where @faissaloo linked. Anyway, I'm going to close this because I believe the original arithmetic error was actually a gcc bug:
https://stackoverflow.com/a/32534477

I've been using unordered_map in a similar way myself for proto2es/emake (with pointers as keys as well as values). But I'm hesitant to accept this as not a gcc bug because the gcc in the portable ZIP (which Harri was likely using) is around 4.8.2 which is 1 minor version older than the 4.9.4 version that StackOverflow reply said no longer reproduces the issue with `std::unordered_map`.

Also it seems like this GCC bug report (that's been fixed in 4.9) seems very relevant:
https://gcc.gnu.org/bugzilla/show_bug.cgi?format=multiple&id=61143

So it does look like this one is resolved and the user's original issue is fully resolved. At the same time, expect that the BasicGUI extension likely needs some additional work, I don't believe Harri was completely finished with it. If you find additional issues with it, or would like help with another issue, please let us know.

For reference, see the Wiki guide about getting a stack trace from gdb for us if you do run onto the issue again: https://enigma-dev.org/docs/Wiki/Debugging

Thanks.
